### PR TITLE
Homework2 task2 - Training Management System - ready

### PR DESCRIPTION
**Pull request by VytasR**
Homework2 task2 - Training Management System - ready for review

### CODE REVIEW

File: Homework2/TrainingManagement/TrainingEntity/Training.cs [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+                    return false;
+                }
+            }
+            return true;
+        }
+
+        // Returns a deep copy of this Training.
+        public Training Clone()
+        {
+            Training other = new Training(new StringBuilder(Description).ToString());
```
Comment (radislavkireev) [inline Homework2/TrainingManagement/TrainingEntity/Training.cs:47]: you do not need to envelope strings to StringBuilder any time, that does not bring benefits.
Replace with `
Training other = new Training(Description);`

Apply similar update to the code below in Clone method.

File: Homework2/TrainingManagement/TrainingEntity/Training.cs [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+            
+            foreach (var lesson in Lessons)
+            {
+                string description = new StringBuilder(lesson.Description).ToString();
+
+                if (lesson is Lecture)
```
Comment (radislavkireev) [inline Homework2/TrainingManagement/TrainingEntity/Training.cs:53]: more elegant solution would be to create Clone() method in lesson and override it in Lecture / PracticalLesson. 
```
class Lesson 
{
   public abstract Lesson Clone();
}
class Lecture : Lesson
{
  public override Lesson Clone() {  / .. Do the cloning code here ... / }
}
```

After that, Training.Clone will look like

```
foreach(var lesson in Lessons)
{
  other.Add(lesson.Clone());
}
```

**Review by radislavkireev — CHANGES_REQUESTED** [review]

File: Homework2/TrainingManagement/TrainingEntity/Training.cs (hunk 1 above)
Comment (VytasR) [reply Homework2/TrainingManagement/TrainingEntity/Training.cs]: corrected

File: Homework2/TrainingManagement/LessonEntities/PracticalLesson.cs [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+using System.Threading.Tasks;
+
+namespace TrainingManagement.LessonEntities
+{
+    internal class PracticalLesson:Lesson
+    {
+        public string LinkToTaskCondition { get; set; }
+        public string LinkToSolution { get; set; }
+
+        public PracticalLesson(string description, string linkToTaskCondition, string linkToSolution):base(description)
```
Comment (radislavkireev) [inline Homework2/TrainingManagement/LessonEntities/PracticalLesson.cs:14]: minor - add spaces for readability
`
 public PracticalLesson(string description, string linkToTaskCondition, string linkToSolution) **:** base(description)`

**Review by radislavkireev — APPROVED** [review]

File: Homework2/TrainingManagement/LessonEntities/PracticalLesson.cs (hunk 3 above)
Comment (VytasR) [reply Homework2/TrainingManagement/LessonEntities/PracticalLesson.cs]: added

**Pull request merged**